src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -75,6 +75,20 @@ internal partial class Binder
                         sourceTuple.HasErrors);
                 }
 
+                // identity stackalloc conversions result in a converted node
+                // to indicate that stackalloc conversions are no longer applicable.
+                // nothing else changes
+                if (source.Kind == BoundKind.StackAllocArrayCreation)
```

Comment:
why would `conversion.IsIdentity` ever be true in case of stackalloc conversions?

Reply:
That would be true in a case  of   `(Span<int>)stackalloc int[10];`

The change is to prevent the following from working:

int * ptr = (Span<int>)stackalloc int[10];

I actually doubt now whether it is needed

---
In reply to: [139542693](https://github.com/dotnet/roslyn/pull/22162#discussion_r139542693) [](ancestors = 139542693)